scripts/tls/gen_cert.sh

```diff
@@ -154,7 +185,13 @@ check_args() {
 case $1 in
     "ca")
         parse_args "$@"
-        create_self_signed "ca"
+        if [ $2 == "rsa" ]; then
+            create_self_signed "ca"
+        elif [ $2 == "dsa" ]; then
```

Comment:
Can you try the following suggested fix?
```suggestion
        if [ $2 = "rsa" ]; then
            create_self_signed "ca"
        elif [ $2 = "dsa" ]; then
```

When running on Ubuntu I got the following errors
```
./scripts/tls/gen_cert.sh: 188: [: rsa: unexpected operator
./scripts/tls/gen_cert.sh: 190: [: rsa: unexpected operator
```

Reply:
@dhurley thank you for your feedback!
I put necessary modification to the script.
I haven't such errors on my _Fedora release 37 (Thirty Seven)_

Reply:
I was debugging test code bu using debugger, no errors were reproduced. Are there any prerequisites before committing the tests, something like run `make deps` and so on? I couldn't find it anywhere.